Core/include/Acts/TrackFitting/KalmanFitter.hpp

```diff
@@ -313,7 +320,10 @@ class KalmanFitter {
              state.navigation.navigationBreak)) {
           // Remove the missing surfaces that occur after the last measurement
           result.missedActiveSurfaces.resize(result.measurementHoles);
-          if (reversedFiltering) {
+          // now get track state proxy for the smoothing logic
+          auto trackStateProxy =
+              result.fittedStates.getTrackState(result.lastTrackIndex);
+          if (m_reverseFilteringLogic(trackStateProxy, reversedFiltering)) {
```

Comment:
I don't know if we still need the boolean `reversedFiltering` if we have the `reverseFilteringLogic`.

Reply:
That's what I was thinking at first but if we do that we lose the option to turn on/off the reverse filtering in the option. In that case we would need to change the reverseFilteringLogic we use and then recompile.

Reply:
So, I kept it in so that we can turn the use of the reversedFiltering on/off in the option without having to recompile with a new reverseFilteringLogic. Let me know if that is fine.

Reply:
OK. Indeed, otherwise, we need to recompile for switching it on/off. Fine with me then.